purchase_partial_invoicing/wizard/po_line_invoice.py

```diff
@@ -20,97 +20,83 @@
+    @api.model
+    def default_get(self, fields):
+        ctx = self.env.context.copy()
+        po_ids = ctx.get('active_ids', [])
+        po_line_obj = self.env['purchase.order.line']
         lines = []
-        for po_line in po_line_obj.browse(cr, uid,
-                                          context.get('active_ids', []),
-                                          context=context):
+        for po_line in po_line_obj.browse(po_ids):
             lines.append({
                 'po_line_id': po_line.id,
                 'product_qty': po_line.product_qty - po_line.invoiced_qty,
                 'invoiced_qty': po_line.product_qty - po_line.invoiced_qty,
                 'price_unit': po_line.price_unit,
             })
-        defaults = super(purchase_line_invoice, self).default_get(
-            cr, uid, fields, context=context)
+        defaults = super(purchase_line_invoice, self).default_get(fields)
         defaults['line_ids'] = lines
         return defaults
 
-    def makeInvoices(self, cr, uid, ids, context=None):
-        if context is None:
-            context = {}
-        ctx = context.copy()
-        wizard = self.browse(cr, uid, ids[0], context=context)
-        purchase_line_obj = self.pool.get('purchase.order.line')
+    @api.multi
+    def makeInvoices(self):
+        self.ensure_one()
+        ctx = self.env.context.copy()
         changed_lines = {}
-        context['active_ids'] = []
-        not_invoiced_ids = []
-        invoiced_ids = []
-        for line in wizard.line_ids:
+        ctx['active_ids'] = []
+        not_invoiced_lines = self.env['purchase.order.line']
+        invoiced_lines = self.env['purchase.order.line']
+        for line in self.line_ids:
             if line.invoiced_qty > line.product_qty:
-                raise orm.except_orm(
-                    _('Error'),
+                raise exceptions.Warning(
                     _("""Quantity to invoice is greater
-                    than available quantity""")
-                )
-            context['active_ids'].append(line.po_line_id.id)
+                    than available quantity"""))
+            ctx['active_ids'].append(line.po_line_id.id)
             changed_lines[
                 line.po_line_id.id
             ] = line.invoiced_qty
             if line.po_line_id.fully_invoiced:
-                invoiced_ids.append(line.po_line_id.id)
+                invoiced_lines += line.po_line_id
             else:
-                not_invoiced_ids.append(line.po_line_id.id)
-        if not_invoiced_ids:
-            purchase_line_obj.write(cr, uid, not_invoiced_ids,
-                                    {'invoiced': False})
-        if invoiced_ids:
-            purchase_line_obj.write(cr, uid, not_invoiced_ids,
-                                    {'invoiced': True})
+                not_invoiced_lines += line.po_line_id
+        if not_invoiced_lines.ids:
+            not_invoiced_lines.write({'invoiced': False})
```

Comment:
You don't need the if because if will not write anything if it is an empty recordset.